Tasks/AzureMonitorAlerts/azurerestutility.ts

```diff
@@ -0,0 +1,265 @@
+                }
+            }
+            else {
+                deferred.reject(tl.loc('CouldnotfetchaccesstokenforAzureStatusCode', response.message.statusCode, response.message.statusMessage));
+            }
+        }, (error) => {
+            deferred.reject(error);
+        });
+
+    return deferred.promise;
+}
+
+export async function getAzureRmResourceDetails(
+    endpoint, 
+    resourceUri: string) : Promise<any> {
+    
+    let deferred: Q.Deferred<any> = Q.defer<any>();
+    let accessToken: string = await getAuthorizationToken(endpoint);
+   
+    // resourceUri is of the form : /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}
+
+    let splittedResourceUri: string[] = resourceUri.split("/");
+    let resourceGroupName: string = splittedResourceUri[4];
+    let resourceType: string = splittedResourceUri[6] + "/" +  splittedResourceUri[7];
+    let resourceName: string = splittedResourceUri[8];
+
+    tl.debug(`Getting AzureRm resource details - '${resourceName}' in resource group '${resourceGroupName}'`);
+
+    let apiVersion = "2017-05-10";
+    let restUrl = `${endpoint.url}subscriptions/${endpoint.subscriptionId}/resourceGroups/${resourceGroupName}/resources?$filter=resourceType EQ '${resourceType}' AND name EQ '${resourceName}'&api-version=${apiVersion}`;
+    let restOptions: IRequestOptions = {
+        additionalHeaders: {
+            "Authorization": "Bearer " + accessToken
+        }
+    };
+
+    restObj.get(restUrl, restOptions)
+        .then((response: IRestResponse<any>) => {
+            if(response.statusCode === 200) {
+                deferred.resolve(response.result["value"][0]);
+            }
+            else {
+                tl.debug(`Getting AzureRm resource '${resourceName}' failed. Status Code: ${response.statusCode}. Response: ${JSON.stringify(response.result, null, 2)}`);
+                deferred.reject(response.statusCode);
+            }
+
+        }, (error) => {
+            deferred.reject(error);
+        });
+
+    return deferred.promise;
+}
+
+export async function getAzureRmAlertRule(
+    endpoint, 
+    resourceGroupName: string, 
+    alertRuleName: string) : Promise<any> {
+
+    let deferred: Q.Deferred<any> = Q.defer<any>();
+    let accessToken: string = await getAuthorizationToken(endpoint);
+
+    tl.debug(`Getting AzureRm alert rule - '${alertRuleName}' in resource group '${resourceGroupName}'`);
+
+    let apiVersion: string = "2016-03-01";
+    let restUrl: string = `${endpoint.url}subscriptions/${endpoint.subscriptionId}/resourceGroups/${resourceGroupName}/providers/Microsoft.insights/alertrules/${alertRuleName}?api-version=${apiVersion}`;
+    let restOptions: IRequestOptions = {
+        additionalHeaders: {
+            "Authorization": "Bearer " + accessToken
+        }
+    };
+
+    restObj.get(restUrl, restOptions)
+        .then((response: IRestResponse<any>) => {
```

Comment:
unlike the app service deploy task, all the rest calls being made here handle the response in the same way. create a common function so that all the responses can be consumed in standard way. You can pass these params to that function: the response, an array of valid statusCode for the promise resolvs, and the loc string to be displayed.

Reply:
The same value is not resolved every time. Hence, abstracting it out to be a common function will only complicate the logic. Further we will need to handle scenarios differently for GET, PUT etc.

Reply:
true, but you can resolve with the result everytime, and then expect the calling side to handle the complete response however it wants to. Or better, pass a processResponse function, and resolve with it's output like so deferred.resolve(processResponse(response.result))

Reply:
you won't need to handle GET,PUT separately. I am asking to change how the response is handled. It will always have a statuscode and result regardless of the kind of call being made